src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -159,23 +234,24 @@ protected override int MakeSlot(BoundExpression node)
             return base.MakeSlot(node);
         }
 
-        private bool IsTrackableAnonymousTypeProperty(PropertySymbol propSymbol)
+        private static bool IsTrackableAnonymousTypeProperty(PropertySymbol propSymbol)
         {
             return !propSymbol.IsStatic &&
                    propSymbol.IsReadOnly &&
                    propSymbol.ContainingType.IsAnonymousType &&
-                   (propSymbol.Type.IsReferenceType || EmptyStructTypeCache.IsTrackableStructType(propSymbol.Type.TypeSymbol));
+                   IsTrackableType(propSymbol.Type);
         }
 
-        private Symbol GetNonFieldSymbol(int slot)
+        private static bool IsTrackableType(TypeSymbolWithAnnotations type)
         {
-            VariableIdentifier variableId = variableBySlot[slot];
-            while (variableId.ContainingSlot > 0)
-            {
-                Debug.Assert(variableId.Symbol.Kind == SymbolKind.Field || (variableId.Symbol.Kind == SymbolKind.Property));
-                variableId = variableBySlot[variableId.ContainingSlot];
-            }
-            return variableId.Symbol;
+            return (object)type != null &&
+                type.IsReferenceType || EmptyStructTypeCache.IsTrackableStructType(type.TypeSymbol);
+        }
+
+        private static bool IsTrackableType(TypeSymbol type)
+        {
+            return (object)type != null &&
+                type.IsReferenceType || EmptyStructTypeCache.IsTrackableStructType(type);
```

Comment:
>type [](start = 83, length = 4)

Same issue here, it looks like we possibly passing null.

Reply:
Fixed.